src/main/java/wtf/meier/tariff/interpreter/model/tariff/Tariff.kt

```diff
@@ -1,38 +1,54 @@
 package wtf.meier.tariff.interpreter.model.tariff
 
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
 import wtf.meier.tariff.interpreter.extension.durationMillis
+import wtf.meier.tariff.interpreter.extension.minus
+import wtf.meier.tariff.interpreter.helper.serializer.CurrencySerializer
+import wtf.meier.tariff.interpreter.helper.serializer.RateIdSerializer
+import wtf.meier.tariff.interpreter.helper.serializer.TariffIdSerializer
+import wtf.meier.tariff.interpreter.helper.serializer.TimeZoneSerializer
 import wtf.meier.tariff.interpreter.model.Interval
+import wtf.meier.tariff.interpreter.model.goodwill.Goodwill
 import wtf.meier.tariff.interpreter.model.rate.Rate
 import wtf.meier.tariff.interpreter.model.rate.RateId
 import java.time.DayOfWeek
 import java.time.Instant
 import java.util.*
 import java.util.concurrent.TimeUnit
 
-
-@Suppress("EXPERIMENTAL_FEATURE_WARNING")
-inline class TariffId(val id: Long)
+// changed from inline to data class, because inline classes are unsupported by kotlin serialization
+data class TariffId(val id: Long)
 
 class InvalidTariffFormatException(message: String) : RuntimeException(message)
 
+@Serializable
 sealed class Tariff {
     abstract val id: TariffId
-    abstract val freeSeconds: Int
     abstract val rates: Set<Rate>
     abstract val billingInterval: Interval?
+    abstract val goodwill: Set<Goodwill>?
+    abstract val currency: Currency
 
 }
 
+@Serializable
+@SerialName("SlotBasedTariff")
 data class SlotBasedTariff(
+    @Serializable(with = TariffIdSerializer::class)
     override val id: TariffId,
-    override val freeSeconds: Int,
     override val rates: Set<Rate>,
     override val billingInterval: Interval?,
+    @Serializable(with = CurrencySerializer::class)
+    override val currency: Currency,
+    override val goodwill: Set<Goodwill>? = null,
```

Comment:
Can you give an example where multiple goodwill make sense in a Tariff?

Reply:
I don't have a good example. There is no good purpose for multiple goodwill and yes, that is a point to remove it not to give any the possibility to abuse it.